src/resources/material.js

```diff
@@ -186,7 +186,8 @@ class MaterialHandler {
 
     _onTextureLoad(parameterName, materialAsset, textureAsset) {
         this._assignTexture(parameterName, materialAsset, textureAsset.resource);
-        materialAsset.resource.update();
+        if (--materialAsset.textureLoadCount === 0)
```

Comment:
The changes in this file look good, but I don't understand it all enough to be confident. I assume _bindAndAssignAssets is called more often when in Editor / live launcher and the material is being edited (textures added / removed), so please test this well. It's tricky to maintain a count in a case like this, when things change.

Reply:
I tested this in the editor with changing the sharing, adding/removing textures and reloading the project. Is there some other test that would better check if this might cause issues?